url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/LookaheadHttpHeader.java

```diff
@@ -0,0 +1,115 @@
+package net.hetimatan.util.http;
+
+
+import java.io.IOException;
+import net.hetimatan.io.file.MarkableReader;
+import net.hetimatan.util.event.EventTaskRunner;
+import net.hetimatan.util.io.ByteArrayBuilder;
+
+public class LookaheadHttpHeader {
+	public static final int EOF = 0;
+	public static final int CRLF = 1;
+	public static final int KEEP = 2;
+	private MarkableReader mCurrentReader = null;
+	private long mStartPointer = 0;
+	private long mStartTime = 0;
+	private boolean mIsFirst = true;
+
+	public LookaheadHttpHeader(MarkableReader reader, int size) throws IOException {
+		mCurrentReader = reader;
+		mStartPointer = reader.getFilePointer();
+		mStartTime = System.currentTimeMillis();
+	}
+
+	public long getStart() {
+		return mStartPointer;
+	}
+
+	public long getElapsedTime() {
+		return System.currentTimeMillis()-mStartTime;
+	}
+
+	public static final byte[] FIRST_LF1= {'\r', '\n'};
+	public static final byte[] FIRST_LF2= {'\n'};
+
+	public static final byte[] LF1= {'\r', '\n', '\r', '\n'};
+	public static final byte[] LF2= {'\r', '\n', '\n'};
+	public static final byte[] LF3= {'\n', '\r', '\n'};
+	public static final byte[] LF4= {'\n', '\n'};
+	
+	private boolean match(byte[] buf, int bufLen, byte[] pat, boolean oneshot) {
+		if(bufLen<pat.length) {
+			return false;
+		}
+		boolean ret = false;
+
+		for(int j=0;j<bufLen;j++) {
+			ret = true;
+			for(int i=0;i<pat.length;i++) {
+				if(buf[i+j]!= pat[i]){
+					ret = false;
+					break;
+				}
+			}
+			if(oneshot||ret==true){break;}
+		}
+		return ret;
+	}
+
+	public int readByEndOfHeader(boolean checkCrlf) throws IOException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.